src/encoded/static/components/experiment.js

```diff
@@ -456,10 +456,21 @@ const ExperimentComponent = ({ context, auditIndicators, auditDetail }, reactCon
     // indicates isogenic.
     const anisogenic = context.replication_type ? (anisogenicValues.indexOf(context.replication_type) !== -1) : false;
 
-    // Get a list of related datasets, possibly filtering on their status.
-    let seriesList = [];
+    // Get a map of related datasets, possibly filtering on their status and
+    // categorized by their type.
+    const seriesMap = {};
     if (context.related_series && context.related_series.length > 0) {
-        seriesList = _(context.related_series).filter(dataset => loggedIn || dataset.status === 'released');
+        context.related_series.forEach(
+            (series) => {
+                if (loggedIn || series.status === 'released') {
+                    if (series['@type'][0] in seriesMap) {
+                        seriesMap[series['@type'][0]].push(series);
+                    } else {
+                        seriesMap[series['@type'][0]] = [series];
+                    }
+                }
+            }
+        );
```

Comment:
I’m not recommending you change this because this works, but for future reference, the underscore library has [one function](http://underscorejs.org/#groupBy) that does exactly this. Inside the callback you would, after the checks, would just return `series['@type'][0]`.

Reply:
Oh. Yes, `groupBy` is what I'm looking for. I stopped when I didn't see useful things on MDN. Thank you for that!